rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSource.java

```diff
@@ -87,24 +101,35 @@ static KeyValueDataSource makeDataSource(@Nonnull Path datasourcePath, @Nonnull
 
         ds.init();
 
+        if (shouldGenerateDbKindFile) {
+            KeyValueDataSource.generatedDbKindFile(kind, fullDbDir);
+        }
```

Comment:
@rmoreliovlabs, it looks like that this doesn't cover scenario when `dbKind.properties` file is present in a root database folder but not in a specific datasource folder (eg. `blocks`, `states` etc.) - the case of current implementation and we need to be backward compatible with that. In this case `dbKind.properties` file should be generated in the specific folder

Reply:
Addressed

Reply:
it looks like the change does not cover the scenario when root `dbKind.properties` exists but specific one does not ( happens when switching from older version to the one that includes changes from this PR )

Reply:
Addressed